api/base/views.py

```diff
@@ -6,57 +6,249 @@
+    to interact with the endpoint.  Unauthorized use will result in 401 Unauthorized or 403 Forbidden responses.  Use
+    `HEAD` to probe an endpoint and make sure your headers are well-formed.  `GET` will return a representation of the
+    entity or entity collection referenced by the endpoint.  An `OPTIONS` request will return a JSON object that describes the
+    endpoint, including the name, a description, the acceptable request formats, the allowed response formats, and any
+    actions available via the endpoint.
+
+    ###Filtering
+
+    Entity collections can be filtered by adding a query parameter in the form:
+
+        filter[<fieldname>]=<matching information>
+
+    String queries are filtered using substring matching. For example, if you were trying to find [Lise
+    Meitner](http://en.wikipedia.org/wiki/Lise_Meitner):
+
+        /users/?filter[full_name]=meitn
+
+    You can filter on multiple fields, or the same field in different ways, by &-ing the query parameters together.
+
+        /users/?filter[full_name]=lise&filter[family_name]=mei
+
+    Boolean fields should be queried with `true` or `false`.
+
+        /nodes/?filter[registered]=true
+
+    ###Pagination
+
+    All entity collection endpoints respond to the `page` query parameter behavior as described in the [JSON-API
+    pagination spec](http://jsonapi.org/format/1.0/#crud).  However, pagination links are provided in the response, and
+    you are encouraged to use that rather than adding query parameters by hand.
+
+    ###Formatting POST/PUT/PATCH request bodies
+
+    The OSF API follows the JSON-API spec for (create and update requests)[http://jsonapi.org/format/#crud].  This means
+    all request bodies must be wrapped with some metadata.  Each request body must be an object with a `data` key
+    containing at least a `type` member.  The value of the `type` member must agree with the `type` of the entities
+    represented by the endpoint.  If not, a 409 Conflict will be returned.  The request should also contain an
+    `attributes` member with an object containing the key-value pairs to be created/updated.  PUT/PATCH requests must
+    also have an `id` key that matches the id part of the endpoint.  If the `id` key does not match the id path part, a
+    409 Conflict error will be returned.
+
+    ####Example 1: Creating a Node via POST
+
+        POST /v2/nodes/
+        {
+          "data": {
+            "type": "nodes",
+            "attributes": {
+              "title" : "A Phylogenetic Tree of Famous Internet Cats",
+              "category" : "project",
+              "description" : "How closely related are Grumpy Cat and C.H. Cheezburger? Is memefulness inheritable?"
+            }
+          }
+        }
+
+    ####Example 2: Updating a User via PUT
+
+        PUT /v2/users/me/
+        {
+          "data": {
+            "id": "3rqxc",
+            "type": "users",
+            "attributes": {
+              "full_name" : "George Lazenby",
```

Comment:
Just a personal agenda item, but could you use female scientists (who are not Marie Curie and, since she's used elsewhere in the docs, Lise Meitner) as user examples? Here's a good list. https://en.wikipedia.org/wiki/List_of_female_scientists_before_the_21st_century